cmd/podman/logout.go

```diff
@@ -65,13 +66,26 @@ func logoutCmd(c *cliconfig.LogoutValues) error {
 		return nil
 	}
 
-	err := config.RemoveAuthentication(sc, server)
-	switch err {
+	// username of user logged in to server (if one exists)
+	userFromAuthFile, passFromAuthFile, err := config.GetAuthentication(sc, server)
+	if err != nil {
+		return errors.Wrapf(err, "error reading auth file")
+	}
+
+	islogin := docker.CheckAuth(getContext(), sc, userFromAuthFile, passFromAuthFile, server)
```

Comment:
Unless I'm misreading the code, CheckAuth returns nil if the credentials would be accepted by the registry and not if the user had already established credentials with the registry.  I don't think it's a valid way to check if the user is or is not logged in with docker login, rather it's a good check to see if they could be.

Reply:
this indicates the valid authentication exists in the machine if `userFromAuthFile != ""&& passFromAuthFile != ""`. So it's alreay logged to the registry. Is it correct?

Reply:
Sorry, slipped on getting back to you on this one.  My understanding is CheckAuth only verifies that the credentials presented are valid.  It does not check to see if the credentials are in use or were accepted.  I could be proven wrong.